### PR TITLE
Implement schema and concepts support in pipeline executor

### PR DESCRIPTION
**Pull request by Dakostu**
This PR implements schema and concepts support for the pipeline executor using the `event_types` and the new `global_concepts` singleton.

### CODE REVIEW

File: libvast/include/vast/global_concepts.hpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2023 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#pragma once
+
+#include "vast/taxonomies.hpp"
+
+namespace vast::global_concepts {
```
[the remark this answers is not in the document]
Comment (Dakostu) [reply libvast/include/vast/global_concepts.hpp]: I implemented a global `modules` registry in https://github.com/tenzir/vast/pull/3081/commits/39a4d079a7285d5457ac6508aec7bfbcc4bd3876

File: libvast/src/system/catalog.cpp [hunk 2]
```diff
@@ -664,6 +665,7 @@ catalog(catalog_actor::stateful_pointer<catalog_state> self,
     self->quit(std::move(err));
     return catalog_actor::behavior_type::make_empty_behavior();
   }
+  global_concepts::init(self->state.taxonomies.concepts);
```
[the remark this answers is not in the document]
Comment (Dakostu) [reply libvast/src/system/catalog.cpp]: I unified the initialization into a singular `modules::init()` call in `vast.cpp` in https://github.com/tenzir/vast/pull/3081/commits/0e596d0f793253eb089918102c11a7da9bdcfb09.

File: libvast/src/pipeline.cpp [hunk 3]
```diff
@@ -40,12 +41,12 @@ class local_control_plane final : public operator_control_plane {
...
-    die("not implemented");
+  auto schemas() noexcept -> const std::vector<type>& override {
+    return vast::modules::schemas();
   }
 
-  auto concepts() const noexcept -> const concepts_map& override {
-    die("not implemented");
+  auto concepts() noexcept -> const concepts_map& override {
+    return vast::modules::concepts();
   }
```
Comment (dominiklohmann) [inline libvast/src/pipeline.cpp:50]: As far as I can tell these can still be marked `const`, as it doesn't access any state.

**Review by dominiklohmann — APPROVED** [review]
Thanks, this is exactly how I imagined it. With that, we should have an easy way for parsing with a known schema in the new framework! 🚀

Just two minor comments left, please address them before merging.

File: libvast/include/vast/modules.hpp [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// Returns an empty list if init(...) was not called.
+auto schemas() -> const std::vector<type>&;
+
+// Get the concepts map.
+// Returns an empty map if init(...) was not called.
+auto concepts() -> const concepts_map&;
+
+// Get the list of modules.
+[[deprecated("call modules::schemas() instead")]] auto global_module()
+  -> const module*;
```
[the remark this answers is not in the document]
Comment (Dakostu) [reply libvast/include/vast/modules.hpp:28]: I fixed the code documentation, but for some reason GitHub doesn't show the updated commits on my side.

File: libvast/src/pipeline.cpp (hunk 3 above)
Comment (Dakostu) [reply libvast/src/pipeline.cpp:50]: I reinstated the east-`const`ness for those methods, but for some reason GitHub doesn't show the updated commits on my side.

Comment (dominiklohmann) [reply libvast/src/pipeline.cpp]: Technical side note:

East or west const refer to whether you want write `const auto` or `auto const` when qualifying a type. The `const` I asked you to add back is not a _const type qualifier_, but rather a _const member function qualfiier_.

**Pull request merged**